others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file
+ * @brief Iterative version of Preorder, Postorder and preorder traversal of tree,
+ (https://en.wikipedia.org/wiki/Tree_traversal).
+ * @author [Motasim](https://github.com/motasimmakki)
+ * @details
+ * 
+ * ### Iterative Preorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current, and pop and elemnt from the stack.
+ * Store the data of current into the result array, and start traversing from it.
+ * Push both the child node of the current node into the stack, first right child then left child.
+ * Repeat the same set of steps untill the Stack becomes empty.
+ * And return the result array as the preorder traversal of a tree.
+ * 
+ * ### Iterative Postorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current, and pop and elemnt from the stack.
+ * Store the data of current into the result array, and start traversing from it.
+ * Push both the child node of the current node into the stack, first left child then right child.
+ * Repeat the same set of steps untill the Stack becomes empty.
+ * Now reverse the result array and then return it to the calling function as a postorder traversal of a tree.
+ * 
+ * ### Iterative Inorder Traversal of a tree
+ * Create a Stack that will store the Node of Tree.
+ * Push the root node into the stack.
+ * Save the root into the variabe named as current.
+ * Now iterate and take the current to the extreme left of the tree by traversing only to its left.
+ * Pop the elemnt from the stack and assign it to the current.
+ * Store the data of current into the result array.
+ * Repeat the same set of steps until the Stack becomes empty or the current becomes NULL.
+ * And return the result array as the inorder traversal of a tree.
+ */
+#include <iostream>   /// for I/O operations
+#include <cstdlib>    /// for `malloc`
+#include <stack>      /// for `stack`
+#include <vector>     /// for `vector`
+#include <algorithm>  /// for `reverse`
+
+// struct that defines the structure of the node of a tree.
+using Node = struct BinaryTree{
+    int data;
+    struct BinaryTree *left;
+    struct BinaryTree *right;
+};
+
+/**
+ * @brief will allocate the memory for a node and, along the data and return the node.
+ * @param data 
+ * @return pointer to the newly created node with assigned data. 
+ */
+Node *createNewNode(int data){
+	auto node = static_cast<struct BinaryTree*>(malloc(sizeof(struct BinaryTree)));

```

Comment:
`malloc()` is a C function. You are not supposed to use this in C++ context. C++ has `new` and `delete` operators try to use them or use `smart pointers` (prefer to use `new` and `delete` in this particluar case)

Reply:
```suggestion
	auto *node = static_cast<Node *>(new (Node));
```

> I tried to replace the use of 'malloc' using 'new' like this and getting 5210 warnings and any of which is not specified.

Reply:
Refactor your code in Class based approach, all your problems will get solved automatically,

Reply:
I also tried the Class-based approach, but it is coming up with the same set of warnings Can these clang-tidy checks could be disabled?

Reply:
Those are not supposed to be disabled. You are facing those errors because you are not managing memory resources properly in your code. Please push your updated code, so that i can see.

Reply:
Yes, I totally agree with you, I tried to references the pointer to the node using the smart pointer but having trouble with that also, I haven't work the smart pointers before and can not afford plagiarism in my code.